[MM-55075] Incorporate new findings into Postgres migration guidelines (#6731)
mattermost/docs: source/deploy/postgres-migration.rst

   DROP INDEX IF EXISTS idx_posts_message_txt;
   DROP INDEX IF EXISTS idx_fileinfo_content_txt;

> Artifacts may remaining from previous configurations/versions

Reviewer: @nab-77 - ``Artefact`` is the British English spelling. ``Artifact`` is US English. All of our docs aim for US English :)

> Artifacts may remaining from previous configurations/versions

Author: Oh, TIL!

> Artifacts may remaining from previous configurations/versions

Reviewer 2: > All of our docs aim for US English

Neil surreptitiously adding British spelling to our docs :smiling_imp:

> Artifacts may remaining from previous configurations/versions

Reviewer 3: I'm just relived that Queen Elizabeth is not here to see this happen ;-)

~~~~~~~~~~~~~~~~~

> Prior to ``v6.4``, Mattermost was using `golang-migrate <https://github.com/golang-migrate/migrate>`__ to handle the schema migrations. Since we don't use it anymore, we exclude the table ``schema_migrations``. If you were using Mattermost before ``v6.4`` consider dropiing this table and exclude this table from comparison as well. 

Reviewer 2: ```suggestion
Prior to ``v6.4``, Mattermost was using `golang-migrate <https://github.com/golang-migrate/migrate>`__ to handle the schema migrations. Since we don't use it anymore, we exclude the table ``schema_migrations``. If you were using Mattermost before ``v6.4`` consider dropping this table and excluding it from comparison as well.
```

Author: Oops, fixing it.

> Prior to ``v6.4``, Mattermost was utilizing `golang-migrate <https://github.com/golang-migrate/migrate>`__ for handling the schema migrations. Since we don't use it anymore, the table ``schema_migrations`` shouldn't be migrated. If you were using Mattermost before ``v6.4`` consider excluding this table from migration and comparison. 

Reviewer: ```suggestion
Prior to ``v6.4``, Mattermost was using `golang-migrate <https://github.com/golang-migrate/migrate>`__ to handle the schema migrations. Since we don't use it anymore, the table ``schema_migrations`` shouldn't be migrated. If you were using Mattermost before ``v6.4`` consider excluding this table from migration and comparison. 
```

.. code:: sql

>    DROP TABLE mattermost.schema_migrations;

Reviewer 2: The "consider excluding this table from migration" sounds a bit weird, since what they need to do is drop it at the source database, right? Maybe something more explicit like "consider dropping that table before migrating"?

Author: Yeah, I agree.  This is actually a kind of FYI, it is already being ignored while executing the migration.

Reviewer 2: Yeah, that's why I thought that "consider excluding this table from migration" is actually not 100% correct, since it'll be already excluded, right? And if it is being excluded, do we really need this?

> Another note about configurations is that; if you were previously utilizing a database for handling the Mattermot configuration, those tables should be removed from your MySQL database. Consider running following DDL to drop tables.

Reviewer 2: ```suggestion
Another note about configurations is that; if you were previously utilizing a database for handling the Mattermost configuration, those tables should be removed from your MySQL database. Consider running following DDL to drop tables.
```

Reviewer: Is our target audience for this content knowledgeable about what ``DDL`` stands for?

Author: I'd say yes if they are executing these queries :) But wouldn't hurt to add extra info I suppose, shall I add?

Reviewer: I recommend expanding out the acronym once, the first time it's referenced on the page, followed by the acronym itself in brackets. You could then use the acronym instead, and readers will know what it stands for.

.. code:: sql

   DROP TABLE ConfigurationFiles;
   DROP TABLE Configurations;
   DROP TABLE db_config_migrations;

> Some community members have reported that they had ``description`` and ``nextsyncat`` columns in their ``SharedChannelRemotes`` table. These columns should be removed from the Table, consider running following DDL to drop the columns. (This migration will be added to future versions of Mattermost)

Reviewer 2: Typo
```suggestion
Some community members have reported that they had ``description`` and ``nextsyncat`` columns in their ``SharedChannelRemotes`` table. These columns should be removed from the table, consider running following DDL to drop the columns. (This migration will be added to future versions of Mattermost)
```

Author: Corrected with @cwarnermm's suggestion.

> Some community members have reported that they had ``description`` and ``nextsyncat`` columns in their ``SharedChannelRemotes`` table. These columns should be removed from the Table, consider running following DDL to drop the columns. (This migration will be added to future versions of Mattermost)

Reviewer: ```suggestion
Some community members have reported that they had ``description`` and ``nextsyncat`` columns in their ``SharedChannelRemotes`` table. These columns should be removed from the table. Consider running following DDL to drop the columns. (This migration will be added to future versions of Mattermost).
```
